tests/ballerina-integration-test/src/test/resources/websub/services/01_websub_publisher.bal

```diff
@@ -227,3 +230,26 @@ function getPayloadContent(string contentType, string mode) returns string|xml|j
     error e = error(websub:WEBSUB_ERROR_CODE, { message : errorMessage });
     panic e;
 }
+
+function checkSubscriberAvailability(string topic, string callback) {
+    boolean subscriberAvailable = false;
+    while(!subscriberAvailable) {
+        websub:SubscriberDetails[] topicDetails = webSubHub.getSubscribers(topic);
+        if (isSubscriberAvailable(topicDetails, callback)) {
+            subscriberAvailable = true;
+        } else {
+            runtime:sleep(1000);
+        }
+    }
```

Comment:
Shall we add maximum loop count to avoid endless loop if subscriber not available. if max loop count reached, break the loop and fail the test case.

Reply:
The test fails when the LogLeecher timeouts and task clears during the server cleanup. So should we add maxLoopCount too?

Reply:
The LogLeecher time out would be after we respond to the request at https://github.com/ballerina-platform/ballerina-lang/pull/13281/files#diff-d319b3fb1a8c4ff418213f19622aaadaR155 right? So the test will fail when HttpClientRequest.doPost times out?

+1 to add a maximum count here instead of relying on timing out.

Reply:
HttpClientRequest timeout is 30s. But the Logleecher timeout is 45s. So the test depends on  Logleecher timeout basically.

However added the MaxCount for safety purposes ;)